Library/Formula/mlton.rb

```diff
@@ -0,0 +1,131 @@
+
+# MLton is a self-hosting compiler for Standard ML.
+# In order to build from source, this formula uses the corresponding
+# upstream binary release to bootstrap.
+
+class Mlton < Formula
+  desc "Whole-program, optimizing compiler for Standard ML"
+  homepage "http://mlton.org"
+  url "https://downloads.sourceforge.net/project/mlton/mlton/20130715/mlton-20130715.src.tgz"
+  version "20130715"
+  sha256 "215857ad11d44f8d94c27f75e74017aa44b2c9703304bcec9e38c20433143d6c"
+
+  depends_on "gmp"
+
+  # The corresponding upstream binary release used to bootstrap.
+  resource "bootstrap" do
+    url "https://downloads.sourceforge.net/project/mlton/mlton/20130715/mlton-20130715-2.amd64-darwin.gmp-static.tgz"
+    sha256 "16a6d4e300f45f4af094692cf8033390e4634fa4c072caf6e9c288234100ad22"
+  end
+
+  # The upstream source release hard codes common locations for GMP;
+  # on darwin, these include '/opt/local' (MacPorts) and '/sw' (Fink).
+  # Replace these with placeholders for the Homebrew GMP location.
+  patch :DATA
```

Comment:
The source package has no `configure` step and the upstream policy has been to include common locations for GMP on various target platforms.  The upstream [installation instructions](http://mlton.org/Installation) explicitly mention editing the `bin/mlton` script to find `gmp.h` and `libgmp.a`.  This patch wouldn't be appropriate for upstream, since it removes support for Fink and MacPorts.  Support for default Homebrew prefix is already implicit upstream, since `-I/usr/local/include` and `-L/usr/local/lib` are used; patching upstream with a placeholder for a non-standard Homebrew prefix seems heavy weight.

I could remove the patch and achieve the same effect with only two `inreplace` lines, at the expense of leaving `/opt/local/{include,lib}` and `/sw/{include,lib}` in the search paths.